impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -412,6 +445,24 @@ private static String ensureFilterHasParens(String filter) {
         }
     }
 
+    private static void processNamingException(NamingException e) throws Exception {
+      String exString = e.toString();
+      if (exString !=null && exString.indexOf("LDAP: error code 48") >= 0) {
```

Comment:
Not sure this is necessary, and it may not be portable. I think it should be possible to just catch `AuthenticationException` -- I'm almost certain that "error code 48" is going to turn out to be `AuthenticationException` anyway, or at least one of the one of the `NamingSecurityExceptions`.

The `NamingException` hierarchy is actually pretty rich -- we should be able to figure out what's going on just by catching the right exceptions, based on the documentation for which methods throw which exceptions under what circumstances. By wrapping and re-throwing the exceptions here, we're actually losing information that would help the higher layers figure out how to respond.